File: _정답코드/3주차/06_두_수의_합_4/Solution.java
public class Solution {
    public int[] solution(int[] numbers, int target) {
        int[] answer = {-1, -1};

        int n = numbers.length;
        int left = 0, right = n-1;

        while(left < right){
            int total = numbers[left] + numbers[right];
            if(total == target){
                answer = new int[] {numbers[left], numbers[right]};
                left ++;
            } else if(total < target){
                left ++;
            } else {
                right --;
            }
        }
        return answer;
    }
}

File: _정답코드/3주차/06_두_수의_합_4/solution.cpp
#include <vector>
using namespace std;

vector<int> solution(vector<int> numbers, int target) {
    vector<int> answer = {-1, -1};
    int n = numbers.size();
    int left = 0, right = n - 1;

    while (left < right) {
        int total = numbers[left] + numbers[right];
        if (total == target) {
            answer = {numbers[left], numbers[right]};
            left++;
        } else if (total < target) {
            left++;
        } else {
            right--;
        }
    }

    return answer;
}